### ISSUE TITLE
Print the values that are being compared in detailed dataframe from `sc.equal()`

### ISSUE THREAD

**Issue by kelvinburke** [sciris/sciris#559]
Hi @cliffckerr 

This issue is a feature request. I was setting up semi-automated testing of optima HIV, where it would run a bunch of projects either updating the version or not and then compare the results before and after. I wanted to compare exactly where in the outputs the results were different - so that if we are just adding an extra results output then it is easy to tell that is the case, and that all the other results are the same. And when results are different, we would like to see how they are different.

So essentially adding the values that are being compared to the output dataframe when `detailed=True`

Calling this would then look something like:
```
import sciris as sc

a = {'x': {'y': 1, 'z': 2}, 'extra': 4}
b = {'x': {'y': 1, 'z': 3}}

df = sc.equal(b,a, detailed=True, output_vals=True)


import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.width', 100)

print(df)
```
with output:
```
          obj0==obj1                                 val0                     val1  equal
root           False  {'x': {'y': 1, 'z': 2}, 'extra': 4}  {'x': {'y': 1, 'z': 3}}  False
(x,)           False                     {'y': 1, 'z': 2}         {'y': 1, 'z': 3}  False
(x, y)          True                                    1                        1   True
(x, z)         False                                    2                        3  False
(extra,)       False                                    4                  MISSING  False
```
(note also I changed `obj0_obj1` to `obj0==obj1` due to personal preference in readability)

I've mocked up an implementation in:
https://github.com/kelvinburke/sciris/blob/output_vals_Equal/sciris/sc_nested.py

A related issue is whether or not the comparison should walk over keys of all objects being compared:
eg. if you swap `a` and `b` in the above example you get the output:
```
        obj0==obj1                     val0                                 val1  equal
root         False  {'x': {'y': 1, 'z': 3}}  {'x': {'y': 1, 'z': 2}, 'extra': 4}  False
(x,)         False         {'y': 1, 'z': 3}                     {'y': 1, 'z': 2}  False
(x, y)        True                        1                                    1   True
(x, z)       False                        3                                    2  False
```
where we don't get to see the `extra` key. Including the `extra` key would should a better look into how the results are different imo.

It shouldn't be too hard to implement this, replacing:
```
bkeys = set(self.bdict.keys()) # Get the base keys (object structure)
```
with something like
```
all_keys = set()
all_keys_list = [all_keys.add(key) or key for dict in self.dicts for key in dict.keys() if key not in all_keys]
```
and then loop over `all_keys_list` instead of `btree`

**Comment by kelvinburke** [comment]
Actually maybe better to do.
```
all_keys = set()
for dict in self.dicts: all_keys = all_keys.union(dict.keys())
all_keys = sorted(all_keys)
```
This way if the second dict has a key that should come before some of the keys in the first then it goes in the correct order. 
I'll leave it up to you to think about the order, I'm not totally sure.

**Comment by cliffckerr** [comment]
While trying to implement this, encountered this: https://github.com/pandas-dev/pandas/issues/16098

**Comment by cliffckerr** [comment]
Closed by #563
